src/librustdoc/html/static/storage.js

```diff
@@ -111,8 +111,18 @@ function switchTheme(styleElem, mainStyleElem, newTheme) {
     });
     if (found === true) {
         styleElem.href = newHref;
-        updateLocalStorage("rustdoc-theme", newTheme);
+        // If this new value comes from a system setting or from the previously saved theme, no
+        // need to save it.
+        if (skipStorage !== true) {
```

Comment:
Could we make this a positive variable instead of `skip`? So that the if could become `if (saveTheme) { ... }`